### PR TITLE
Filter the fdb entries

### PR DESCRIPTION
**Pull request by niusmallnan**
https://github.com/rancher/rancher/issues/9822

### CODE REVIEW

File: vxlan/fdb_utils.go [hunk 1]
```diff
@@ -18,8 +18,10 @@ func getCurrentFDBEntries(link netlink.Link) (map[string]*netlink.Neigh, error)
 
 	fdbEntries := make(map[string]*netlink.Neigh)
 	for index, n := range neighs {
-		log.Debugf("getCurrentFDBEntries: Neigh %+v", n)
-		fdbEntries[n.IP.To4().String()] = &neighs[index]
+		if n.IP != nil {
+			log.Debugf("getCurrentFDBEntries: Neigh %+v", n)
+			fdbEntries[n.IP.To4().String()] = &neighs[index]
```
Comment (yasker) [inline vxlan/fdb_utils.go:23]: I don't understand how the problem gets fixed. 

I think if `n.IP` is nil, then `n.IP.To4().String()` should exploded with NPE exception?

Comment (niusmallnan) [reply vxlan/fdb_utils.go:23]: @yasker 
see this: https://play.golang.org/p/6VH-6wq8tx

Comment (yasker) [reply vxlan/fdb_utils.go:23]: https://golang.org/pkg/net/#IP

`type IP []byte`

It turns out you can call functions of nil array in Go. Interesting.

Comment (yasker) [reply vxlan/fdb_utils.go:23]: This is even more interesting. 

It turns out `nil` can be used as a key in `map[string]xxx`.

https://play.golang.org/p/H5uGE6HoWQ

**Comment by yasker** [comment]
So what's happened is, one of the entry will be deleted by `updateFDB` logic, is due the combination of two things: n.IP.To4.String() return `nil` rather than NPE; and map in Golang can use `nil` as key for `string` type key. `updateFDB` will delete the old entry because there is no `ip == nil` to be found in the new entry list.

LGTM now.

**Pull request merged**